### PR TITLE
chore(docs, examples): #147 - Added documentation regarding `Status`

### PR DESCRIPTION
**Pull request by grzesuav**
Signed-off-by: grzesuav <grzesuav@gmail.com>

### CODE REVIEW

**Review by AmitKumarDas — DISMISSED** [review]

File: docs/src/guide/best-practices.md [hunk 1]
```diff
@@ -101,3 +101,46 @@ For example, if you observe 2 Pods, but you return a desired list of 3 Pods,
 you should return a Status that reflects only the observed Pods
 (e.g. `replicas: 2`).
 This is important so that Status reflects present reality, not future desires.
+
+#### Working with Status subresource in metacontroller
+ If you would like to use `Status` subresource, you should take care about:
```
Comment (fpetkovski) [inline docs/src/guide/best-practices.md:106]:
```suggestion
 If you would like to expose and use the `Status` subresource in your custom resource, you should take care of:
```

File: docs/src/guide/best-practices.md [hunk 2]
```diff
@@ -101,3 +101,46 @@ For example, if you observe 2 Pods, but you return a desired list of 3 Pods,
...
+ 1. proper CRD schema definition for `Status` section, in order to let metacontroller
```
Comment (fpetkovski) [inline docs/src/guide/best-practices.md:107]:
```suggestion
 1. having a proper CRD schema definition for `Status` section in order to let metacontroller
```

File: docs/src/guide/best-practices.md [hunk 3]
```diff
@@ -101,3 +101,46 @@ For example, if you observe 2 Pods, but you return a desired list of 3 Pods,
...
+ to update it successfully - it must be a part of CRD schema, i.e.
```
Comment (fpetkovski) [inline docs/src/guide/best-practices.md:108]:
```suggestion
 update it successfully - it must be a part of CRD schema, i.e.
```

File: docs/src/guide/best-practices.md [hunk 4]
```diff
@@ -101,3 +101,46 @@ For example, if you observe 2 Pods, but you return a desired list of 3 Pods,
...
+               actual_copies:
+                 type: integer
+               observedGeneration:
+                 type: integer
+         required:
+         - spec
+     subresources:
+       status: {}
+ ```
+2. your controller must be strict about schema defined in CRD, i.e., in example above
```
Comment (fpetkovski) [inline docs/src/guide/best-practices.md:141]:
```suggestion
2. your controller must be strict about the types in the schema defined in CRD, i.e., in example above
```

File: docs/src/guide/best-practices.md [hunk 5]
```diff
@@ -101,3 +101,46 @@ For example, if you observe 2 Pods, but you return a desired list of 3 Pods,
...
+do not try to set any of the fields as `string` or add additional fields there.
```
Comment (fpetkovski) [inline docs/src/guide/best-practices.md:142]:
```suggestion
do not try to set any of the `integer` fields as `string`s, or add additional fields there.
```

**Comment by grzesuav** [comment]
thanks @AmitKumarDas @fpetkovski for comments !

**Pull request merged**